shared/src/main/scala/implicits.scala

```diff
@@ -17,16 +17,15 @@
 package fetch
 
 import cats.{Eval, MonadError}
-import cats.std.FutureInstances
+import cats.instances.FutureInstances
 import scala.concurrent.{Promise, Future, ExecutionContext}
 
 object implicits extends FutureInstances {
   implicit def fetchFutureFetchMonadError(
-      implicit ec: ExecutionContext,
-      ME: MonadError[Future, Throwable]
+      implicit ec: ExecutionContext
   ): FetchMonadError[Future] = new FetchMonadError[Future] {
     override def runQuery[A](j: Query[A]): Future[A] = j match {
-      case Sync(e) => ME.pureEval(e)
+      case Sync(e) => pure(e.value)
```

Comment:
Thanks for taking the time to make these changes Ilya, I've made some changes to the fetch execution reporting that should make those errors go away in the `reporting` branch. I'm going to try to combine your work and mine and cut a release as soon as possible.